cmd/template/cluster/provider/common.go

```diff
@@ -27,7 +27,20 @@ type ClusterCRsConfig struct {
 	EKS                bool
 	ExternalSNAT       bool
 	ControlPlaneSubnet string
-	PodsCIDR           string
+
+	// OpenStack only.
```

Comment:
Are all of these options really unique per workload cluster? 
Wouldn't it make way more sense to default a bunch of these (e.g. SSHKeyName, ImageName, ...) in the webhook as they are tied to the management cluster?

Additionally we know that customers want to use gitops to manage these clusters, supplying them with good overlays seems more sensible to me.

Reply:
I dropped SSH key name because we don't want to allow it. Image name will be decided by us (we're managing images), but is dependent on Kubernetes version and may be useful to override. Source UUID is tricky because it depends on the OpenStack project the cluster is deployed into and that is therefore cluster-dependent.

For the alpha (or "pre-alpha") we just want the ability to specify all of the options. The flags won't show up in the automatic CLI help so I don't think it will be a problem in terms of backwards compatibility. Let me know if you disagree @MarcelMue.